script.js

```diff
@@ -1,12 +1,125 @@
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      makePageForEpisodes(data);
+    });
+
+  let allShows = getAllShows();
+  //Sort the Series
+  allShows.sort(function(a, b) {
+    if (a.name < b.name) {
+      return -1;
+    }
+    if (a.name > b.name) {
+      return 1;
+    }
+    // names must be equal
+    return 0;
+  });
+
+  const showList = document.querySelector("#series-list");
+  showList.innerHTML = seriesSelector(allShows);
+
+  showList.addEventListener("change", function (event) {
+    const showId = event.target.value;
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        makePageForEpisodes(data);
+      });
+  });
 }
 
+//content create
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let episodes = document.querySelector(".episodeContainer");
+  episodes.innerHTML = createNewList(episodeList);
+  
+  const getInputField = document.querySelector(".search-episodes");
+
+  // Dropdown rolling Series
+  const dropDownSearchMenu = document.querySelector("#episode-list");
+  dropDownSearchMenu.addEventListener("change", function (event){
+    const episodeId = event.target.value;
+    const episodesFilteredById = episodeList.filter(
+    (episode) => episode.id == episodeId
+    );
+    episodes.innerHTML = createNewList(episodesFilteredById);
+  });
+
+  dropDownSearchMenu.innerHTML = createDropDownMenu(episodeList);
+  
+  //Search Button function and counter
+  getInputField.addEventListener("keyup", function () {
+    
+      filteredEpisodes = episodeList.filter(
+      (episode) =>
+        episode.summary.toLowerCase().includes(getInputField.value.toLowerCase()) ||
+        episode.name.toLowerCase().includes(getInputField.value.toLowerCase())
+    );
+    episodes.innerHTML = createNewList(filteredEpisodes);
+    let searchCounter = document.querySelector(".counter")
+    searchCounter.textContent = `${filteredEpisodes.length} episode(s)`  
+  });
+}
+
+//Episodes Drop Down list
+function createDropDownMenu(episodeList) {
+  return episodeList
+    .map(function (item) {
+      return `<option value =${item.id}>
+      S${item.season.toString().padStart(2,"0")}
+      E${item.number.toString().padStart(2, "0")} 
+      ${item.name}</option>`;
+    })
+  .join("");
+}
+
+//Episodes creation 
+function  createNewList(episodeList){
+  let searchCounter = document.querySelector(".counter")
+  searchCounter.textContent = `${episodeList.length} episode(s)`  
+return episodeList.map(function (item) {
+    return `<div class="episode">
+    <h1 class="episodeHeader">${item.name} - S${item.season.toString().padStart(2, "0")}E${item.number.toString().padStart(2, "0")}</h1>
+    <img src=${item.image.medium} alt= ${item.name}>
```

Comment:
Breaking this down into smaller functions would make this easier to read. For example

```
getImageElement(item ){
   return <img src=${item.image.medium} alt= ${item.name}>
}
```

Reply:
For this one, I think I should start using document.createElement().to create elements rather than use Strings . Then the page could respond faster and I could be able to use smaller functions . Thanks